packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -73,36 +113,68 @@ export class RouterliciousRestWrapper extends RestWrapper {
 		const translatedConfig = this.useRestLess ? this.restLess.translate(config) : config;
 		const fetchRequestConfig = axiosRequestConfigToFetchRequestConfig(translatedConfig);
 
-		const response: Response = await this.rateLimiter.schedule(async () =>
-			fetch(...fetchRequestConfig).catch(async (error) => {
+		const res = await this.rateLimiter.schedule(async () => {
+			const perfStart = performance.now();
+			const result = await fetch(...fetchRequestConfig).catch(async (error) => {
 				// Browser Fetch throws a TypeError on network error, `node-fetch` throws a FetchError
 				const isNetworkError = ["TypeError", "FetchError"].includes(error?.name);
 				throwR11sNetworkError(
 					isNetworkError ? `NetworkError: ${error.message}` : safeStringify(error),
 				);
-			}),
-		);
+			});
+			return {
+				response: result,
+				duration: performance.now() - perfStart,
+			};
+		});
 
+		const response = res.response;
+
+		let start = performance.now();
+		const text = await response.text();
+		const receiveContentTime = performance.now() - start;
+
+		const bodySize = text.length;
+		start = performance.now();
 		const responseBody: any = response.headers.get("content-type")?.includes("application/json")
-			? await response.json()
-			: await response.text();
+			? JSON.parse(text)
+			: text;
+		const parseTime = performance.now() - start;
 
 		// Success
 		if (response.ok || response.status === statusCode) {
-			const result: T = responseBody;
-			return result;
+			const result = responseBody;
+			if (addNetworkCallProps) {
+				const headers = headersToMap(response.headers);
+				return {
+					content: result,
+					headers,
+					requestUrl: fetchRequestConfig[0].toString(),
+					propsToLog: {
+						...getPropsToLogFromResponse(headers),
+						bodySize,
+						receiveContentTime,
+						parseTime,
+						callDuration: res.duration,
+					},
+				} as unknown as T;
```

Comment:
this casting looks fishy. It's too easy to make mistake - here (miss some properties that are required), or by caller, as type differs substantially based on addNetworkCallProps.
I think we should always return IR11sResponse<T> and let the caller remove / ignore extra properties if it does not want to.

Any reason why we want to make it optional?